website/templates/project/project_header.mako

```diff
@@ -106,8 +106,16 @@
 
         % if not node['is_retracted']:
            % if not node['is_pending_registration']:
-              <div class="alert alert-info">This ${node['node_type']} is a registration of <a class="link-solid" href="${node['registered_from_url']}">this ${node['node_type']}</a>; the content of the ${node['node_type']} has been frozen and cannot be edited.</div>
-
+                % if not file_name:
+                    <div class="alert alert-info">This ${node['node_type']} is a registration of <a class="link-solid" href="${node['registered_from_url']}">this ${node['node_type']}</a>; the content of the ${node['node_type']} has been frozen and cannot be edited.</div>
+                % else:
+                    % if urls.get('archived_from'):
+                        <div class="alert alert-info">This file is part of a registration and is being shown in its archived version (and cannot be altered).
+                            The <a class="link-solid" href="${urls['archived_from']}">active file</a> is viewable from within the <a class="link-solid" href="${node['registered_from_url']}">live ${node['node_type']}</a>.</div>
+                    % else:
+                        <div class="alert alert-info">This ${node['node_type']} is a registration of <a class="link-solid" href="${node['registered_from_url']}">this ${node['node_type']}</a>; the content of the ${node['node_type']} has been frozen and cannot be edited.</div>
```

Comment:
Since this is the same as above, is there any way to modify this logic so that you don't have to repeat this?